### ISSUE TITLE
Large initial data insertion is slower with Observable enabled

### ISSUE THREAD

**Issue by CodePirate7** [dexie/Dexie.js#1787]
When initializing the database and bulk loading a large amount of initial data (e.g. 20,000+ rows), there is a noticeable performance hit with Observables enabled. The initial insertion takes around 15 seconds with Observables on, but reduces to 10 seconds when Observables are disabled.
This is likely because each insertion triggers Observable notifications and events. For a large initial population, these events become expensive.
A suggestion would be to allow selectively disabling Observables during initial bulk loading, but keeping them enabled for incremental updates later.
For example, allow something like:
```js
// Load initial data 
db.disableObservables();
db.bulkPut(largeDataset); 

// Future updates
db.enableObservables();
```
This would provide a bulk loading performance boost while retaining Observables for normal usage.
An alternative could be a bulkPut() option to disable Observables just during that bulk operation.
In summary, allowing a way to opt-out of Observables during initial population could improve performance when bulk loading many rows into a new database. But retain Observables for incremental updates later for reactivity.
Let me know if any clarification would be helpful! Overall loving the reactivity Observables enable in Dexie.``

**Comment by dfahlander** [comment]
Thanks for bringing this up. I suppose this is about `liveQuery()` observables and not the legacy `Dexie.Observable`?

Currently, observables are triggered after a transaction completes, so in practice a large transaction would be silent until it completes. But the normal and scalable use case is that data loading must be done out of several transactions (such as a stream from a server resulting into a number of bulkPut calls), this would trigger observables many times. This is a typical case for first-time sync of offline-first apps.

The situation can be avoided though in front-end code by making sure that the app doesn't show any components with live queries while bulk loading. For example, set the application state into a state where the root component only renders a progress bar or so with explanatory text about things are being loaded.

What do you think? Are there situations I don't think of, making it easier with pausable liveQuery observables than to unmount components that use liveQueries? Maybe liveQuery observables that aren't from a front-end component? My gut feeling is that there might be situations where your suggestion is valid...
